cmd/allocator/main.go

```diff
@@ -244,7 +251,7 @@ func main() {
 		os.Exit(0)
 	})
 
-	h := newServiceHandler(ctx, kubeClient, agonesClient, health, conf.MTLSDisabled, conf.TLSDisabled, conf.remoteAllocationTimeout, conf.totalRemoteAllocationTimeout, conf.allocationBatchWaitTime)
```

Comment:
I have used `codes.Code` instead of `http` following your suggestion. [However, in the latest commit, I've commented out the `http` code usage, here I just tried to use the `http` code and converting it into gRPC using the `GrpcCodeFromHTTPStatus` converter method.]
I am looking forward to hearing your thoughts...

Reply:
I am missing the `GrpcCodeFromHTTPStatus` call?

This looks mostly like what I meant, from about here down. I think you were initially correct that the config surface (the flag, the helm variable) should be in http status, though, since most of the use of changing this is actually on the http side. So in this file (allocator/main.go), I would take a flag for unallocated http status and convert it into a codes.Code. Then the rest looks reasonable.